plugins/dependency-graph-resolver/src/main/java/eu/f4sten/depgraph/Main.java

```diff
@@ -15,55 +15,163 @@
  */
 package eu.f4sten.depgraph;
 
+import static eu.fasten.core.maven.resolution.MavenResolverIO.simplify;
+import static eu.fasten.core.utils.MemoryUsageUtils.logMemoryUsage;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Set;
+
 import javax.inject.Inject;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import eu.f4sten.depgraph.data.Coordinates;
-import eu.f4sten.depgraph.endpoints.Hello;
 import eu.f4sten.infra.Plugin;
 import eu.f4sten.infra.http.HttpServer;
 import eu.f4sten.infra.json.TRef;
 import eu.f4sten.infra.kafka.DefaultTopics;
 import eu.f4sten.infra.kafka.Kafka;
 import eu.f4sten.infra.kafka.Message;
-import eu.f4sten.pomanalyzer.data.PomAnalysisResult;
+import eu.f4sten.infra.utils.IoUtils;
+import eu.fasten.core.maven.data.Pom;
+import eu.fasten.core.maven.resolution.MavenResolverData;
 
 public class Main implements Plugin {
 
     private static final Logger LOG = LoggerFactory.getLogger(Main.class);
+    private static final int NUM_TO_REPORT = 1000;
+
+    // TODO this thresholds still need tweaking
+    private static final int MIN_STORAGE_NUMBER = 10000;
```

Comment:
Maybe you can introduce a CLI arg for the `MIN_STORAGE_NUMBER` threshold with a default value.

Reply:
I think this is a good idea, I will do that.

Reply:
Fixed in the latest commit. I have also updated the numbers based on my own experience after playing around with this on monster.